[FLINK-16811][jdbc] introduce row converter API to JDBCDialect

bowenli86 opened the pull request:
## What is the purpose of the change

commit 1: make StringUtils.arrayToString() convert array of byte array correctly
commit 2: [FLINK-16811][jdbc] introduce row converter API to JDBCDialect

## Brief change log

## Verifying this change
UT and IT added

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

bowenli86 commented:
@wuchong can you take a look?

wuchong commented on flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/source/row/converter/JDBCRowConverter.java:
```diff
@@ -0,0 +1,39 @@
+	 * @param resultSet ResultSet from JDBC
+	 * @param reuse The row to set
+	 */
+	Row setRow(ResultSet resultSet, Row reuse) throws SQLException;
```
What about to call the method `convert`? It's not very explict that it happens some conversion in `setRow`.

bowenli86 commented:
> Thanks @bowenli86 . The changes looks good to me. Only left a minor comment about the method name.
> 
> Btw please have a look at the failed build.

Thanks for your review! @wuchong Addressed feedback and will merge this once CI is green
